assembl/graphql/post.py

```diff
@@ -753,25 +755,37 @@ def mutate(root, args, context, info):
             if not post:
                 continue
 
-            new_extract = models.Extract(
-                creator_id=user_id,
-                owner_id=user_id,
-                discussion_id=discussion_id,
-                body=extract.get('body'),
-                important=extract.get('important', False),
-                content=post,
-                extract_nature=extract_nature,
-                extract_state=extract_state
-            )
-            new_extract.lang = extract.get('lang')
-            post.db.add(new_extract)
-            range = models.TextFragmentIdentifier(
-                extract=new_extract,
-                xpath_start=extract.get('xpath_start'),
-                offset_start=extract.get('offset_start'),
-                xpath_end=extract.get('xpath_end'),
-                offset_end=extract.get('offset_end'))
-            post.db.add(range)
-            post.db.flush()
+            db = post.db
+            extract_hash = models.Extract.get_extract_hash(
+                extract.get('lang'),
+                extract.get('xpath_start'),
+                extract.get('xpath_end'),
+                extract.get('offset_start'),
+                extract.get('offset_end'),
+                post_id,
+                extract_nature)
```

Comment:
Are you sure about extract_nature, shouldn't it be extract_nature.value? It will maybe give another hash on app restart if it's the instance of enum entry.